File: Java/deep_bark/src/main/java/com/deepbark/entity/MixDog.java
package com.deepbark.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "mix_dogs")
@Getter
@Setter
public class MixDog {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name_en", nullable = false)
    private String nameEn;

    @Column(name = "name_ko", nullable = false)
    private String nameKo;

    @Column(name = "breed1", nullable = false)
    private String breed1;

    @Column(name = "breed2", nullable = false)
    private String breed2;
} 

File: App_flutter/lib/models/mix_dog.dart
class MixDog {
  final String id;
  final String nameEn;
  final String nameKo;
  final String breed1;
  final String breed2;
  final double confidence;

  MixDog({
    required this.id,
    required this.nameEn,
    required this.nameKo,
    required this.breed1,
    required this.breed2,
    this.confidence = 0.0,
  });

  MixDog copyWith({
    String? id,
    String? nameEn,
    String? nameKo,
    String? breed1,
    String? breed2,
    double? confidence,
  }) {
    return MixDog(
      id: id ?? this.id,
      nameEn: nameEn ?? this.nameEn,
      nameKo: nameKo ?? this.nameKo,
      breed1: breed1 ?? this.breed1,
      breed2: breed2 ?? this.breed2,
      confidence: confidence ?? this.confidence,
    );
  }
} 